streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -833,7 +829,24 @@ public void close() {
      */
     @Deprecated
     public synchronized boolean close(final long timeout, final TimeUnit timeUnit) {
-        log.debug("Stopping Streams client with timeoutMillis = {} ms.", timeUnit.toMillis(timeout));
+        long timeoutMs = timeUnit.toMillis(timeout);
+
+        if (timeoutMs < 0) {
+            timeoutMs = 0;
+        } else if (timeoutMs == 0) {
+            timeoutMs = Long.MAX_VALUE;
+        }
+
+        return close(timeoutMs);
+    }
+
+    /**
+     * @param timeoutMs  how long to wait for the threads to shutdown.
+     * @return {@code true} if all threads were successfully stopped&mdash;{@code false} if the timeout was reached
+     * before all threads stopped
+     */
+    private synchronized boolean close(final long timeoutMs) {
```

Comment:
both public methods are `synchronized` already -- don't think we need to `synchronize` this one.

Nit: we don't need JavaDocs for `private` methods